scipy/stats/_axis_nan_policy.py

```diff
@@ -380,7 +380,7 @@ def result_to_tuple(res):
     if not callable(too_small):
         def is_too_small(samples, *ts_args, **ts_kwargs):
             for sample in samples:
-                if len(sample) <= too_small:
+                if sample.size <= too_small or len(sample) <= too_small:
```

Comment:
I don't think this is right. This is for reducing functions that work along an axis, so why would the `size` of the array be the important thing? 

Before this PR, `is_too_small` when each element of `samples` is 1D - that's why using `len` is OK. If we want to generalize that condition to N-D arrays, it would be `if sample.shape[axis] <= too_small`.

Reply:
Until now, by the time `is_too_small` was called, the decorator had moved all the computation axes to the zeroth axis. So, `len(samples)` actually did the right thing. The only exception to this is my usage in this PR. So, I think you are right, we would just need to add an axis argument and look if the slice in that axis is too small.

Reply:
I think by this point the decorator has moved the computation to the *last* axis:
https://github.com/scipy/scipy/blob/1a68159580eef7e2c8e780e331406615007f0142/scipy/stats/_axis_nan_policy.py#L473

Reply:
Ah, that's right. Since we pass 1-D slices when `is_too_small` was called, everything worked.